### ISSUE TITLE
App opens in fullscreen

### ISSUE THREAD

**Issue by sohojmanush** [ganeshrvel/openmtp#45]
App opens in fullscreen on mac os mojave.

**Comment by ganeshrvel** [comment]
@sohojmanush need more info

**Comment by sohojmanush** [comment]
![Screen-Recording-2019-11-07-at-06 31 55](https://user-images.githubusercontent.com/5019527/68355267-ef21fd80-0138-11ea-809a-4834e1d4774b.gif)

**Comment by LePhi0711** [comment]
Yep, I can confirm this behaviour. OpenMTP-2.4.1 installed on MacOS 10.14.6.
OpenMTP starts using the complete screen which seems a little bit much on a 27" iMac ;-)
The screenshot sohojmanush posted shows exactly what is happening on my machine.

But in fact: as long OpenMTP isn't able to establish a connection to my Garmin Fenix 6 (which is running Android and working fine with AndroidFileTransfer) I can't use OpenMTP and I don't care about this bug ;-)

**Comment by MkStevo** [comment]
Just to also confirm I'm having this happen. I'm running Mojave and when started, OpenMTP fills the screen - covering my (auto-hidden) menu bar. Covering the menu bar makes re-sizing OpenMTP a little difficult as the menu keeps popping up over OpenMTP.

**Comment by ilplaymo** [comment]
confirmed also in Big Sur 11.0.1
here is default window size, too big in my Samsung 28' 4K

![Schermata 2021-01-10 alle 04 04 41](https://user-images.githubusercontent.com/2071355/104113938-f6293a80-52fe-11eb-813b-578fe2d526df.jpg)



for additional information i also attach detailed structure of my machine
 
[HacOS_config.txt](https://github.com/ganeshrvel/openmtp/files/5792185/HacOS_config.txt)

**Comment by escalinci** [comment]
This is my only issue with openmtp, in combination with the UI in my opinion being a bit too big. It doesn't remember the window size when last opened, and always takes up the largest amount of screen space possible. I downloaded the beta to see if this was also the behaviour in that version.

https://user-images.githubusercontent.com/10742076/156920542-b2585bae-73a8-4dd6-9452-e3dac850ddf6.mov

Technically speaking, it's not opening in full screen as that is a separate mode, and your menubar and dock would be obscured.

**Comment by ganeshrvel** [comment]
I will take a look at it.

**Comment by CodyJung** [comment]
I did a little research here and I see why it's opening maximized (calling `mainWindow.maximize()` will do that), but it seems like Electron apps won't preserve their window position without quite a bit of work.

I attempted to put something together for that, but I don't understand settings.js well enough yet to know how to persist and restore settings. And as some other developers on the Internet have noted, there are a lot of edge cases: if you fix it so it works correctly with multiple monitors, you can end up creating a situation where the window is offscreen and you can't get it back. Electron really just needs to create an API for this (there's an 8-year-old feature request ticket for it).

@ganeshrvel I can remove the line that maximizes the window on start, but are you OK with the new behavior where the window always starts as a centered 880x640px window?

**Comment by ganeshrvel** [comment]
The reason I havent removed maximize() by default is that on a smaller screen the file names gets crushed down (on list view) and few other reasons.

What we could here is to write back window dimensions to the settings.json file on a resize event and later when they open the app preset the window size (fetch it from settings.json if it is set).

**Comment by Commodore64user** [comment]
I have a 27" monitor, this stuff opens and uses the full area of the screen, please let the application remember it's last size and honour it.
